### PR TITLE
Cleanup animation docs.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -895,26 +895,25 @@ class Animation:
 
     Parameters
     ----------
-    fig : `~matplotlib.figure.Figure`
-       The figure object that is used to get draw, resize, and any
-       other needed events.
+    fig : `~.Figure`
```
Comment (timhoffm) [inline lib/matplotlib/animation.py]: It is our current policy to use fully qualified names for argument types, see https://matplotlib.org/devdocs/devel/documenting_mpl.html#referencing-types.

(multiple occurrences throught the PR)

File: lib/matplotlib/animation.py [hunk 2]
```diff
@@ -1392,26 +1391,20 @@ class TimedAnimation(Animation):
...
-    interval : number, optional
-       Delay between frames in milliseconds.  Defaults to 200.
-
-    repeat_delay : number, optional
+    fig : `~.Figure`
+        The figure object that is used to get draw, resize, and any
+        other needed events.
+    interval : int, default: 200
+        Delay between frames in milliseconds.
+    repeat_delay : int, default: None
```
Comment (timhoffm) [inline lib/matplotlib/animation.py]:
```suggestion
    repeat_delay : int, default: 0
```
That's what it semantically does. AFAICS we could even use 0 as the actual default value.

File: lib/matplotlib/animation.py [hunk 3]
```diff
@@ -1392,26 +1391,20 @@ class TimedAnimation(Animation):
...
         If the animation in repeated, adds a delay in milliseconds
```
Comment (timhoffm) [inline lib/matplotlib/animation.py]:
```suggestion
         If the animation is repeated, adds a delay in milliseconds
```
But while we're at it, could be written as
> The delay between subsequent runs of the animation, if it is repeated.

File: lib/matplotlib/animation.py [hunk 4]
```diff
@@ -1474,31 +1467,23 @@ class ArtistAnimation(TimedAnimation):
...
-
-    interval : number, optional
-       Delay between frames in milliseconds.  Defaults to 200.
-
-    repeat_delay : number, optional
+        Each list entry is a collection of artists that are made visible on
+        the corresponding frame.  Other artists are made invisible.
+    interval : int, default: 200
+        Delay between frames in milliseconds.
+    repeat_delay : int, default: None
```
Comment (timhoffm) [inline lib/matplotlib/animation.py]: See `repeat_delay` comments above.

File: lib/matplotlib/animation.py [hunk 5]
```diff
@@ -1586,47 +1571,44 @@ def gen_function() -> obj
...
 
-    interval : number, optional
-       Delay between frames in milliseconds.  Defaults to 200.
+    interval : int, default: 200
+        Delay between frames in milliseconds.
 
-    repeat_delay : number, optional
-       If the animation in repeated, adds a delay in milliseconds
-       before repeating the animation.  Defaults to *None*.
+    repeat_delay : int, default: None
```
Comment (timhoffm) [inline lib/matplotlib/animation.py]: See `repeat_delay` comments above.

File: lib/matplotlib/animation.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/animation.py]: I don't agree with that policy, but let's put that discussion to another time.  Reverted.

File: lib/matplotlib/animation.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/animation.py]: agreed (modulo wording)

File: lib/matplotlib/animation.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/animation.py]: agreed.

File: lib/matplotlib/animation.py [hunk 6]
```diff
@@ -1393,25 +1392,19 @@ class TimedAnimation(Animation):
...
-
-    repeat : bool, optional
-        Controls whether the animation should repeat when the sequence
-        of frames is completed.  Defaults to ``True``.
-
-    blit : bool, optional
-        Controls whether blitting is used to optimize drawing.  Defaults
-        to ``False``.
-
+        The figure object that is used to get draw, resize, and any
```
Comment (NelleV) [inline lib/matplotlib/animation.py]: Nitpick. That sentence is awkward. How about:

- `The figure object used to get needed events, such as draw or resize.`

(I'm still unhappy about the word `get`)

File: lib/matplotlib/animation.py [hunk 7]
```diff
@@ -1475,30 +1468,22 @@ class ArtistAnimation(TimedAnimation):
     Parameters
     ----------
     fig : `~matplotlib.figure.Figure`
-       The figure object that is used to get draw, resize, and any
-       other needed events.
-
+        The figure object that is used to get draw, resize, and any
```
Comment (NelleV) [inline lib/matplotlib/animation.py]: ditto

File: lib/matplotlib/animation.py [hunk 8]
```diff
@@ -1550,23 +1535,23 @@ class FuncAnimation(TimedAnimation):
     Parameters
     ----------
     fig : `~matplotlib.figure.Figure`
-       The figure object that is used to get draw, resize, and any
-       other needed events.
+        The figure object that is used to get draw, resize, and any
```
Comment (NelleV) [inline lib/matplotlib/animation.py]: ditto

File: lib/matplotlib/animation.py [hunk 9]
```diff
@@ -1586,47 +1571,44 @@ def gen_function() -> obj
...
+        Whether blitting is used to optimize drawing.  Note: when using
+        blitting, any animated artists will be drawn according to their zorder;
+        however, they will be drawn on top of any previous artists, regardless
+        of their zorder.
 
-    cache_frame_data : bool, optional
-       Controls whether frame data is cached. Defaults to *True*.
-       Disabling cache might be helpful when frames contain large objects.
+    cache_frame_data : bool, default: True
+        Controls whether frame data is cached.
```
Comment (NelleV) [inline lib/matplotlib/animation.py]: nitpick as well. You removed `Controls` for the blit argument. It would make sense to remove it here as well.

**Comment by anntzer** [comment]
Agreed with all.  Additionally also slightly reworded AbstractMovieWriter.setup and inherited MovieWriter.setup from it.

**Comment by NelleV** [comment]
Thanks @anntzer !

**Pull request merged**